src/array/cpu/segment_reduce.h

```diff
@@ -101,6 +103,58 @@ void ScatterAdd(NDArray feat, NDArray idx, NDArray out) {
   }
 }
 
+/*!
+ * \brief CPU kernel of Scatter Add (on first dimension) operator for heterogeneous graph.
+ * \note math equation: out[idx[i], *] += feat[i, *]
+ * \param feat The input tensor.
+ * \param idx The indices tensor.
+ * \param out The output tensor.
+ */
+
+template <typename IdType, typename DType>
+void UpdateGradMinMax_hetero(HeteroGraphPtr graph,
+                       std::vector<NDArray> list_feat,
+                       std::vector<NDArray> list_idx,
+                       std::vector<NDArray> list_idx_ntypes,
+                       std::vector<NDArray> list_out) {
+  std::vector<std::vector<dgl_id_t>> dst_src_ntids(graph->NumVertexTypes(),
+    std::vector<dgl_id_t>());
+  for (dgl_type_t etype = 0; etype < graph->NumEdgeTypes(); ++etype) {
+    auto pair = graph->meta_graph()->FindEdge(etype);
+    const dgl_id_t dst_id = pair.first;  // graph is reveresed
+    const dgl_id_t src_id = pair.second;
+    dst_src_ntids[dst_id].push_back(src_id);  // can have duplicates. Use Hashtable to optimize.
+  }
+  std::vector<bool> updated(graph->NumVertexTypes(), false);
+  for (int dst_id = 0; dst_id < dst_src_ntids.size(); ++dst_id) {
+    updated[dst_id] = false;
```

Comment:
Why only `dst_id`'s flag gets cleared?

Reply:
Does the previous explanation clarifies this question?

Reply:
Addressed and fixed.